### PR TITLE
LPS-165624 Merge available languages for layout default experience

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

See https://issues.liferay.com/browse/LPS-165624.

Layouts have two sets of available languages: those of the layout itself (used for the title and friendly urls, among other things), and those used by its fragments.

When requesting the set of available languages for the default experience we need to include both sets of languages together so that:

1. When translating the page, we accurately display all translated languages.

2. We can display all translations of the friendly urls for that layout.

# How do I tests this?

Pleas follow the steps described in the ticket.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia This is my proposed solution to the friendly url issues. It happened that there was another bug (this time in translations): If you translate the page title to, e.g., es_ES and jp_JP, and the fragments to, say, es_ES and pt_PT, when translating the page jp_JP wouldn't show up in the set of available languages.

Please take a look and if you're fine with this we can move it forward. Thank!

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa two things

What are the other issue LPS
and
on the previous PR about this I made this change :https://github.com/liferay-lima/liferay-portal/pull/3648/commits/3789704e487b162ba3618fcb24388511fd15f381

It's worth to cherry- pick here?

Thanks

**Review by AliciaGarciaGarcia — COMMENTED** [review]
@adolfopa two things

What are the other issue LPS
and
on the previous PR about this I made this change :https://github.com/liferay-lima/liferay-portal/commit/3789704e487b162ba3618fcb24388511fd15f381

It's worth to cherry- pick here?

Thanks

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia I'll cherry pick that commit and push-force here later.

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia Sorry. Can you do the cherry pick? I don't have your branch locally and cannot fetch it right now.

**Comment by AliciaGarciaGarcia** [comment]
Hi @adolfopa 

The commit I mentioned changes the behavior of the component
https://github.com/liferay-lima/liferay-portal/pull/3648#issuecomment-1296746354
https://github.com/liferay-lima/liferay-portal/pull/3648#issuecomment-1300355730

can you confirm this will be the  correct behavior?


Thanks!

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia If it changes behavior (i.e. breaks tests) maybe it is better to send it in a different ticket so that we don't delay this fix.
Are you ok with these changes? If so, let me know and we can forward it.

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/info/item/provider/LayoutInfoItemLanguagesProvider.java [hunk 1]
```diff
@@ -38,12 +43,24 @@
 	public String[] getAvailableLanguageIds(Layout layout)
 		throws PortalException {
 
+		Set<String> availableLanguageIds = new TreeSet<>(
+			Comparator.naturalOrder());
+
+		if (layout.isTypeContent()) {
```
[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/info/item/provider/LayoutInfoItemLanguagesProvider.java:49]: @adolfopa I tried the pr and I have some concerns

[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/info/item/provider/LayoutInfoItemLanguagesProvider.java:49]: This is the correct behavior. In the default experience, the layout title is displayed as another field, so we need to take its translations into account when displaying the available languages. I explained why the previous behavior was a problem in 154c2c467deacf0bfa5c0dd6625cdee18c4ff52f

[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/info/item/provider/LayoutInfoItemLanguagesProvider.java:49]: wow, too much info on a commit message XD

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**